packages/shared/src/contexts/BuyCoresContext.tsx

```diff
@@ -98,165 +80,28 @@ export const BuyCoresContextProvider = ({
     id: string;
     value: number;
   }>();
-  const [paddle, setPaddle] = useState<Paddle>();
-  const isCheckoutOpenRef = React.useRef(false);
   const logRef = useRef<typeof logEvent>();
   logRef.current = logEvent;
 
-  const { data: prices } = useQuery(
-    transactionPricesQueryOptions({
-      user,
-      isLoggedIn,
-    }),
-  );
+  const { paddle, openCheckout } = usePaddlePayment({
+    successCallback: (event) => {
+      setActiveStep({
+        step: SCREENS.PROCESSING,
+        providerTransactionId: event.data.transaction_id,
+      });
+    },
+    targetType: TargetType.Credits,
+  });
+  const { data: prices } = useProductPricing({
+    type: ProductPricingType.Cores,
+  });
 
   const getQuantityForPriceFn = ({ priceId }: { priceId: string }) => {
     return getQuantityForPrice({ priceId, prices });
   };
   const getQuantityForPriceRef = useRef(getQuantityForPriceFn);
   getQuantityForPriceRef.current = getQuantityForPriceFn;
 
-  useEffect(() => {
```

Comment:
This useEffect is managed within the `usePaddlePayment` that is used by both Plus and Cores payment.